File: src/main/java/com/aries/tutorial/adapter/TransactionAdapter.java
package com.aries.tutorial.adapter;

import com.aries.extension.data.TransactionData;
import com.aries.extension.handler.TransactionHandler;
import com.aries.extension.util.PropertyUtil;
import com.aries.extension.util.LogUtil;
import com.aries.tutorial.util.AdapterFormatter;

public class TransactionAdapter implements TransactionHandler {
    @Override
    public void on(TransactionData[] transactions) {
        String subject = PropertyUtil.getValue("transaction_adapter", "subject", "Unknown subject");
        LogUtil.info("[TransactionAdapter] - " + subject + " (transactions=" + transactions.length + ")");

        int idx = 0;
        for (TransactionData data : transactions) {
            idx++;
            LogUtil.info(AdapterFormatter.formatTransaction(idx, data));
        }
    }
}

File: src/main/kotlin/com/aries/tutorial2/adapter/TransactionAdapter.kt
package com.aries.tutorial2.adapter

import com.aries.extension.data.TransactionData
import com.aries.extension.handler.TransactionHandler
import com.aries.extension.util.LogUtil
import com.aries.extension.util.PropertyUtil
import com.aries.tutorial2.util.AdapterFormatter

class TransactionAdapter : TransactionHandler {
    override fun on(transactions: Array<TransactionData>) {
        val subject = PropertyUtil.getValue("transaction_adapter", "subject", "Unknown subject")
        LogUtil.info("[TransactionAdapter] - $subject (transactions=${transactions.size})")

        transactions.forEachIndexed { index, data ->
            LogUtil.info(AdapterFormatter.formatTransaction(index + 1, data))
        }
    }
}
